### PR TITLE
feat(testlib): added global tests rng and waku2 sub-module

### PR DESCRIPTION
**Pull request by LNSD**
This PR improves the current `testlib` module. Some code (e.g., the `generateKey` proc added by @alrevuelta) from the `test_helpers` module has been copied to `testlib` to centralize it.

> **Note**
> All `asyncTest`s not using the `await` keyword, have been changed to `test`.

- [x] Added testlib's `waku2` module:
    + Added key generation functions (e.g., `generateEcdsaKey`, `generateSecp256k1Key`, etc.).
    + Moved into `waku2` module the `fakeWakuMessage` helper proc.
    + Moved into `waku2` module the `newTestSwitch` helper proc.
- [x] Centralized the Random Number Generator instance. All tests requiring an RNG instance should import the `testlib/common` module.
- [x] Updated all test suites' imports.
- [x] Remove the `tests/v2/utils.nim` file (not used).

### CODE REVIEW

File: tests/v2/testlib/waku2.nim [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+proc generateEcdsaKey*(): PrivateKey =
+  PrivateKey.random(ECDSA, rng[]).get()
+
+proc generateEcdsaKeyPair*(): KeyPair =
+  KeyPair.random(ECDSA, rng[]).get()
+
+proc generateSecp256k1Key*(): PrivateKey =
+  PrivateKey.random(Secp256k1, rng[]).get()
+
+proc generateKey*(): PrivateKey =
```
Comment (alrevuelta) [inline tests/v2/testlib/waku2.nim:24]: Perhaps remove `generateKey()` and use generateEcdsaKey and generateSecp256k1Key?

File: tests/v2/waku_relay/test_wakunode_relay.nim [hunk 2]
```diff
@@ -67,11 +65,11 @@ procSuite "WakuNode - Relay":
 
   asyncTest "Messages are correctly relayed":
     let
-      nodeKey1 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      nodeKey1 = generateKey()
```
Comment (alrevuelta) [inline tests/v2/waku_relay/test_wakunode_relay.nim:68]: In general `nodeKeyxx` is never used so perhaps.

`node1 = WakuNode.new(newEcdsaKey(), ValidIpAddress.init("0.0.0.0"), Port(60410))`

see also other tests.
 just a suggestion though.

File: tests/v2/testlib/waku2.nim [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  let peerAddr = address.get(MultiAddress.init("/ip4/127.0.0.1/tcp/0").get())
+  return newStandardSwitch(some(peerKey), addrs=peerAddr)
+
+
+# Waku message
+
+export
+  waku_message.DefaultPubsubTopic,
+  waku_message.DefaultContentTopic
+
```
Comment (rymnc) [inline tests/v2/testlib/waku2.nim:38]: Unsure what's the policy on exports in the middle of the file, what are your thoughts?

**Review by alrevuelta — APPROVED** [review]
nice! some minor non blocking comments.

File: tests/v2/test_enr_utils.nim [hunk 4]
```diff
@@ -39,12 +40,12 @@ procSuite "ENR utils":
...
       empty.toMultiAddresses().len() == 0
 
-  asyncTest "Init ENR for Waku Usage":
+  test "Init ENR for Waku Usage":
     # Tests RFC31 encoding "happy path"
     let
       enrIp = ValidIpAddress.init("127.0.0.1")
       enrTcpPort, enrUdpPort = Port(61101)
-      enrKey = wakuenr.crypto.PrivateKey.random(Secp256k1, rng[])[]
+      enrKey = generateSecp256k1Key()
```
Comment (rymnc) [inline tests/v2/test_enr_utils.nim:48]:
```suggestion
      enrKey = generateKey()
```
As per other usages of `crypto.PrivateKey.random(Secp256k1, rng[])[]`
no strong opinion here though, I like that it's explicit with the curve now

**Review by rymnc — APPROVED** [review]
LGTM, couple queries!

File: tests/v2/testlib/waku2.nim (hunk 1 above)
Comment (LNSD) [reply tests/v2/testlib/waku2.nim:24]: Honestly, this was to simplify the PR. But, as you wish 🤷🏼‍♂️

File: tests/v2/waku_relay/test_wakunode_relay.nim (hunk 2 above)
Comment (LNSD) [reply tests/v2/waku_relay/test_wakunode_relay.nim:68]: I did a find/replace. What you suggest is a ton of work (note how many times these methods are used). 

I suggest that every time one touches a test suite changes this issue 🙂

**Review by jm-clius — APPROVED** [review]
Thanks!

File: tests/v2/waku_relay/test_wakunode_relay.nim (hunk 2 above)
Comment (alrevuelta) [reply tests/v2/waku_relay/test_wakunode_relay.nim]: sounds good thanks, resolving

**Pull request merged**